plugins/vulnerable-chain-finder/src/main/java/eu/f4sten/vulchainfinder/utils/DatabaseUtils.java

```diff
@@ -0,0 +1,121 @@
+import eu.fasten.core.data.vulnerability.Vulnerability;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import org.jooq.DSLContext;
+import org.jooq.Record;
+
+public class DatabaseUtils {
+
+    private final DSLContext context;
+    private final JsonUtils jsonUtils;
+
+    public DatabaseUtils(DSLContext context, JsonUtils jsonUtils) {
+        this.context = context;
+        this.jsonUtils = jsonUtils;
+    }
+
+    public DSLContext getContext() {
+        return context;
+    }
+
+    public MetadataDao getDao(DSLContext ctx) {
+        return new MetadataDao(ctx);
+    }
+
+    public Map<FastenURI, List<Vulnerability>> selectVulCallablesOf(final Set<Long> depIds) {
+        Map<FastenURI, List<Vulnerability>> vulCallables = new HashMap<>();
+        for (Long depId : depIds) {
+            vulCallables = selectVulCallablesOf(depId);
+        }
+        return vulCallables;
+    }
+
+    public Map<FastenURI, List<Vulnerability>> selectVulCallablesOf(final long depId) {
+
+        final var moduleIds = selectAllModulesOf(depId);
+        if (moduleIds == null || moduleIds.isEmpty()) {
+            return Collections.emptyMap();
+        }
+        return selectConcurrentlyVulCallablesOf(moduleIds);
+    }
+
+    public Map<FastenURI, List<Vulnerability>> selectConcurrentlyVulCallablesOf(
+        final Set<Long> moduleIds) {
+        Map<FastenURI, List<Vulnerability>> result = new ConcurrentHashMap<>();
+        moduleIds.parallelStream().forEach(moduleId ->
+            context.fetch(createStrForSelectVulCallablesWhereModuleIdIs(moduleId))
+                .forEach(record -> {
+                    final var vulMap = convertRecordToVulMap(record);
+                    if (!vulMap.isEmpty()) {
+                        result.put(createFastenUriFromPckgVersionUriFields(record),
+                            new ArrayList<>(vulMap.values()));
+                    }
+                }));
+        return result;
+    }
+
+    private Map<String, Vulnerability> convertRecordToVulMap(final Record record) {
+        final var vulField = record.get(3);
+        if (vulField == null) {
+            return Collections.emptyMap();
+        }
+        final var setType = new TRef<HashMap<String, Vulnerability>>() {};
+        return jsonUtils.fromJson(vulField.toString(), setType);
+    }
+
+    public static FastenURI createFastenUriFromPckgVersionUriFields(final Record record) {
+        final var uriString =
+            String.format("%s%s$%s%s",
+                "fasten://mvn!", record.get(0), record.get(1), record.get(2));
+        return FastenURI.create(uriString);
+    }
+
+    public static String createStrForSelectVulCallablesWhereModuleIdIs(final Long moduleId) {
+        return "SELECT packages.package_name, package_versions.version, callables.fasten_uri, " +
```

Comment:
This query no longer works given that we do not store `vulnerability` in the metadata field of the `callables` table. The following SQL query should work:
```sql
SELECT packages.package_name, package_versions.version, callables.fasten_uri
FROM callables, modules, package_versions, packages, vulnerabilities_x_package_versions, vulnerabilities_x_callables
where modules.id = callables.module_id
and package_versions.id = modules.package_version_id
and packages.id = package_versions.package_id
and vulnerabilities_x_package_versions.package_version_id = package_versions.id
and vulnerabilities_x_package_versions.vulnerability_id = vulnerabilities_x_callables.vulnerability_id
and callables.id = vulnerabilities_x_callables.callable_id
and callables.module_id = [MODULE_ID];
```